### PR TITLE
fix(@rjsf/core): Support ui:orderable and ui:removable

### LINKED ISSUE #2256: ui:orderable and ui:removable array options not working
### Prerequisites

- [x] I have read the [documentation](https://react-jsonschema-form.readthedocs.io/);
- [x] In the case of a bug report, I understand that providing a [SSCCE](http://sscce.org/) example is tremendously useful to the maintainers.
- [x] Ideally, I'm providing a [sample JSFiddle](https://jsfiddle.net/n1k0/f2y3fq7L/6/) or a [shared playground link](https://rjsf-team.github.io/react-jsonschema-form/) demonstrating the issue.

### Description

According to the documentation,
```js
"ui:orderable": false,
"ui:removable": false,
"ui:addable" : false
```
should be equivalent to
```js
"ui:options": {
  "orderable": false,
  "removable": false,
  "addable": false
}
```
However, in the first form, only the last line (`"ui:addable"`) seems to be applied, whereas the first two (`"ui:orderable"` and `"ui:removable"`) are not.

### Steps to Reproduce

You can try this by modifying any of the arrays in the "Arrays" example in the Playground.

### Version

Version 2.4.2, using the core package

### PR DESCRIPTION
**Pull request by sophstad**
### Reasons for making this change

Fixes #2256. `orderable` and `removable` worked as designed with the following syntax:
```js
{
  options: {
    orderable: false,
    removable: false
  }
}
```

But not with this syntax:
```js
{
  "ui:orderable": false,
  "ui:removable": false
}
```

This is my first contribution, so let me know if I can do anything else to facilitate review. Thanks for your work on RJSF!

### Checklist

* [ ] **I'm updating documentation**
  - [ ] I've [checked the rendering](https://react-jsonschema-form.readthedocs.io/en/latest/#contributing) of the Markdown text I've added
* [x] **I'm adding or updating code**
  - [ ] I've added and/or updated tests
  - [ ] I've updated [docs](https://react-jsonschema-form.readthedocs.io/) if needed
  - [x] I've updated the [changelog](https://github.com/rjsf-team/react-jsonschema-form/blob/master/CHANGELOG.md) with a description of the PR
* [ ] **I'm adding a new feature**
  - [ ] I've updated the playground with an example use of the feature

### CODE REVIEW

File: packages/core/src/components/fields/ArrayField.js [hunk 1]
```diff
@@ -803,11 +803,7 @@ class ArrayField extends Component {
     const {
       fields: { SchemaField },
     } = registry;
-    const { orderable, removable } = {
-      orderable: true,
-      removable: true,
-      ...uiSchema["ui:options"],
-    };
+    const { orderable = true, removable = true } = getUiOptions(uiSchema);
```
Comment (epicfaace) [inline packages/core/src/components/fields/ArrayField.js:806]: Thank you! Can you add a test to ensure this doesn't regress? Thanks.

File: CHANGELOG.md [hunk 2]
```diff
@@ -17,7 +17,13 @@ should change the heading of the (upcoming) version to include a major version b
 -->
 # v5.0.0 (coming soon)
 
-# v4.1.1 (upcoming)
+# v4.1.2 (upcoming)
+
+## @rjsf/core
+
+- Fix `ui:orderable` and `ui:removable` in arrays
```
Comment (epicfaace) [inline CHANGELOG.md:24]:
```suggestion
- Fix `ui:orderable` and `ui:removable` in arrays (#2797)
```

**Comment by sophstad** [comment]
Hey there @epicfaace @jacqueswho, just wanted to bump this and see if there's anything else needed for a review. Thanks again!

**Review by jacqueswho — APPROVED** [review]

**Comment by epicfaace** [comment]
@sophstad Please fix lint.

**Pull request merged**